examples/twittermap/web/public/javascripts/searchbar/controllers.js

```diff
@@ -4,7 +4,64 @@ angular.module('cloudberry.util', ['cloudberry.common'])
       if ($scope.keyword && $scope.keyword.trim().length > 0) {
         cloudberry.parameters.keywords = $scope.keyword.trim().split(/\s+/);
         // skip the empty query for now.
-        cloudberry.query(cloudberry.parameters);
+
+          //stopword filtering feature
+          String.prototype.removeStopWords = function() {
+              var x;
+              var y;
+              var word;
+              var stop_word;
+              var regex_str;
+              var regex;
+              var cleansed_string = this.valueOf();
+              var stop_words = external_stop_words();
+
+              // Split out all the individual words in the phrase
+              var words = cleansed_string.match(/[^\s]+|\s+[^\s+]$/g);
+
+              // Review all the words
+              for(x=0; x < words.length; x++) {
+                  // For each word, check all the stop words
+                  for(y=0; y < stop_words.length; y++) {
+                      // Get the current word
+                      word = words[x].replace(/\s+|[^a-z]+/ig, "");   // Trim the word and remove non-alpha
+
+                      // Get the stop word
+                      stop_word = stop_words[y];
+
+                      // If the word matches the stop word, remove it from the keywords
+                      if(word.toLowerCase() === stop_word) {
+                          // Build the regex
+                          regex_str = "^\\s*"+stop_word+"\\s*$";      // Only word
+                          regex_str += "|^\\s*"+stop_word+"\\s+";     // First word
+                          regex_str += "|\\s+"+stop_word+"\\s*$";     // Last word
+                          regex_str += "|\\s+"+stop_word+"\\s+";      // Word somewhere in the middle
+                          regex = new RegExp(regex_str, "ig");
+
+                          // Remove the word from the keywords
+                          cleansed_string = cleansed_string.replace(regex, " ");
+                      }
+                  }
+              }
+              return cleansed_string.replace(/^\s+|\s+$/g, "");
+          };
+
+          var rawData = $scope.keyword;
+
+          var myData = rawData.removeStopWords();
+
+          if (myData.valueOf() === "" || null) {
+
+              alert("Invalid input! Please enter again.");
```

Comment:
We can be more kind here.  "Your query only contains stopwords. Please re-enter your query."